### ISSUE TITLE
Viber messages formatted differently now.

### ISSUE THREAD

**Issue by YacoubKureh** [aDrz/analysis-viber-logs#1]
Hello,

I enjoyed your blog post immensely, so I wanted to do the same thing and I saw you had all of the code ready to use. However, I downloaded my messages yesterday and the file I received was formatted as such:

="17/08/2014"   12:52:56    ="Me"   ="+194XXXXXXXXX"    "Hi"
="17/08/2014"   12:58:34    ="Her"  ="01XXXXXXXXXXX"    "Hey"

I've never used R before so there's been a bit of a learning curve trying to change some of the lines in main to get this to work. 

One change was the read in pattern, since mine is tab-delimited I switched the pattern to: "\s_(.+)\t\s_(.+)\t\s_(.+)\t\s_(.+)[3,9]\t\s*(.+)")  However I think having all these quotation marks " " makes things very confusing and I was wondering if you had that in yours and, if so, what you did about it? I just cleared the whole csv of them, but I know that within some messages I used quotation marks so I lost those for now. The other concern of mine are these extra equal signs, I had to make the edit mutate(sender = replace(sender, sender == "=Me", "Him")) 

Also, I was confused about the column names: "date","time","sender","receiver","msg"  as the fourth column is the sender's number for me. Maybe viber changed it. Regardless, I don't think it's used in the code, so it's a non-issue.

Cheers

**Comment by adrz** [comment]
Hello,

Indeed your csv file is quite different than mine. 
I suggest that first you remove all those "=" in the file with:

> sed -i 's/=//g' viber-log.csv

Then in R:

> df <- read.csv("viber-log.csv", header=FALSE, sep=" ",colClasses = "character")
> colnames(df) <- c("date","time","sender","receiver","msg")

Now simply ignore lines 15 to 20 in main.R !

Hope this trick will help

**Comment by adrz** [comment]
I hope only one thing... That you never used double quote " while you were talking ! We'll figure out something help if so

**Comment by YacoubKureh** [comment]
So I messaged my girlfriend and asked her to download her file and hers was
formatted as yours was so I was able to return to using your original code.
However, I found that she use a lot of commas and new lines when typing
since she uses viber on her computer often and sends long messages.
Ultimately, I changed your regex a bit to fix the comma issue, but I just
went through and deleted all 100+ of the new lines but I wasn't sure if
there was a workaround for that since this basically ruins the main premise
of the readlines command.

Also I changed the time zone by adding tz when inputting  

df <- df %>%
  mutate(posix.date=parse_date_time(paste0(date,time),"%d%m%y%H%M%S"),tz="Europe/London") %>%  
  mutate(nb.char = nchar(msg)) %>%
  select(posix.date, sender, msg, nb.char) %>%
  arrange(as.numeric(posix.date))

and then using the following to adjust it

df$posix.date <- as.POSIXct(as.integer(df$posix.date),
                            origin="1970-01-01", 
                            tz="American/Los_Angeles")

This answer came from jlhoward on stackoverflow.
